packages/aws-cdk-lib/aws-ecs/lib/log-drivers/aws-log-driver.ts

```diff
@@ -106,6 +116,10 @@ export class AwsLogDriver extends LogDriver {
     if (props.logGroup && props.logRetention) {
       throw new Error('Cannot specify both `logGroup` and `logRetentionDays`.');
     }
+
+    if (props.maxBufferSize !== '' && props.mode === AwsLogDriverMode.BLOCKING) {
```

Comment:
I think we need to also check if `props.mode` is undefined here. Since `mode` is optional it might not be provided which will cause the default mode of blocking. So maybe it would be cleaner for us to validate based on `maxBufferSize` being provided and `mode` not being `NON_BLOCKING`. Note: I'm basing this off of us making `maxBufferSize` a `number` instead of a `string`

```ts
if (props.maxBufferSize !== undefined && props.mode !== AwsLogDriverMode.NON_BLOCKING) {
  // code
}
```